lang/cs/Org.Apache.REEF.Bridge.Core.Client/Grpc/ClientService.cs

```diff
@@ -0,0 +1,237 @@
+{
+    internal sealed class ClientService : BridgeClient.BridgeClientBase, IClientService
+    {
+
+        private static readonly Logger Log = Logger.GetLogger(typeof(ClientService));
+
+        private static readonly Void Void = new Void();
+
+        private readonly IObserver<ISubmittedJob> _submittedJobHandler;
+
+        private readonly IObserver<IRunningJob> _runningJobHandler;
+
+        private readonly IObserver<ICompletedJob> _completedJobHandler;
+
+        private readonly IObserver<IFailedJob> _failedJobHandler;
+
+        private readonly IObserver<IJobMessage> _jobMessageHandler;
+
+        private readonly IObserver<IFailedRuntime> _failedRuntimeHandler;
+
+        private readonly IObserver<IWakeError> _wakeErrorHandler;
+
+        private REEFClient.REEFClientClient _clientStub = null;
+
+        private string _jobId = "unknown";
+
+        [Inject]
+        private ClientService(
+            [Parameter(Value = typeof(SubmittedJobHandler))] IObserver<ISubmittedJob> submittedJobHandler,
+            [Parameter(Value = typeof(RunningJobHandler))] IObserver<IRunningJob> runningJobHandler,
+            [Parameter(Value = typeof(CompletedJobHandler))] IObserver<ICompletedJob> completedJobHandler,
+            [Parameter(Value = typeof(FailedJobHandler))] IObserver<IFailedJob> failedJobHandler,
+            [Parameter(Value = typeof(JobMessageHandler))] IObserver<IJobMessage> jobMessageHandler,
+            [Parameter(Value = typeof(FailedRuntimeHandler))] IObserver<IFailedRuntime> failedRuntimeHandler,
+            [Parameter(Value = typeof(WakeErrorHandler))] IObserver<IWakeError> wakeErrorHandler)
+        {
+            _submittedJobHandler = submittedJobHandler;
+            _runningJobHandler = runningJobHandler;
+            _completedJobHandler = completedJobHandler;
+            _failedJobHandler = failedJobHandler;
+            _jobMessageHandler = jobMessageHandler;
+            _failedRuntimeHandler = failedRuntimeHandler;
+            _wakeErrorHandler = wakeErrorHandler;
+            LauncherStatus = LauncherStatus.InitStatus;
+        }
+
+        public bool IsDone => LauncherStatus.IsDone;
+
+        public LauncherStatus LauncherStatus { get; private set; }
+
+        public void Close()
+        {
+            if (_clientStub != null)
+            {
+                try
+                {
+                    _clientStub.DriverControlHandler(new DriverControlOp()
+                    {
+                        JobId = _jobId,
+                        Operation = DriverControlOp.Types.Operation.Close
+                    });
+                }
+                catch (Exception e)
+                {
+                    Log.Log(Level.Warning, "exception occurred when trying to close job", e);
+                }
+
+                LauncherStatus = LauncherStatus.ForceCloseStatus;
+                _clientStub = null;
+            }
+        }
+
+        public void Close(byte[] message)
```

Comment:
maybe just
```C#
public void Close(byte[] message = null)
{
    try
    {
        _clientStub?.DriverControlHandler(new DriverControlOp
        {
            JobId = _jobId,
            Message = message != null ? ByteString.CopyFrom(message) : ByteString.Empty,
            Operation = DriverControlOp.Types.Operation.Close
        });
    }
    catch (Exception e)
    {
        Log.Log(Level.Warning, "Exception when trying to close job: " + _jobId, e);
    }

    LauncherStatus = LauncherStatus.ForceCloseStatus;
    _clientStub = null;
}
```
instead of both methods
(and have only `void Close(byte[] message = null)` in the `IClientService` interface?)